cobbler/modules/managers/isc.py

```diff
@@ -225,36 +215,215 @@ def write_dhcp_file(self):
+
+                # add references to the system, profile, and distro
+                # for use in the template
+                if system.name in blender_cache:
+                    blended_system = blender_cache[system.name]
+                else:
+                    blended_system = utils.blender(self.api, False, system)
+                    blender_cache[system.name] = blended_system
+
+                interface["filename"] = blended_system.get("filename")
+                interface["netboot_enabled"] = blended_system["netboot_enabled"]
+                interface["hostname"] = blended_system["hostname"]
+                interface["owner"] = blended_system["name"]
+                interface["name_servers"] = blended_system["name_servers"]
+                interface["mgmt_parameters"] = blended_system["mgmt_parameters"]
+
+                # Explicitly declare filename for other (non x86) archs as in DHCP discover package mostly the
+                # architecture cannot be differed due to missing bits...
+                if distro is not None and not interface.get("filename"):
+                    if distro.arch == "ppc":
+                        interface["filename"] = "grub/grub.ppc"
+                    elif distro.arch == "ppc64":
+                        interface["filename"] = "grub/grub.ppc64"
+                    elif distro.arch == "ppc64le":
+                        interface["filename"] = "grub/grub.ppc64le"
+                    elif distro.arch == "aarch64":
+                        interface["filename"] = "grub/grubaa64.efi"
+
+                if not self.settings.always_write_dhcp_entries:
+                    if not interface["netboot_enabled"] and interface['static']:
+                        continue
+
+                if dhcp_tag == "":
+                    dhcp_tag = blended_system.get("dhcp_tag", "")
+                    if dhcp_tag == "":
+                        dhcp_tag = "default"
+
+                if dhcp_tag not in dhcp_tags:
+                    dhcp_tags[dhcp_tag] = {
+                        mac: interface
+                    }
+                else:
+                    dhcp_tags[dhcp_tag][mac] = interface
+
+        # Remove macs from redundant slave interfaces from dhcp_tags otherwise you get duplicate ip's in the installer.
+        for dt in list(dhcp_tags.keys()):
+            for m in list(dhcp_tags[dt].keys()):
+                if m in ignore_macs:
+                    del dhcp_tags[dt][m]
+
+        # we are now done with the looping through each interface of each system
+        metadata = {
+            "date": time.asctime(time.gmtime()),
+            "next_server_v6": self.settings.next_server,
+            "dhcp_tags": dhcp_tags
+        }
+
+        if self.logger is not None:
+            self.logger.info("generating %s" % self.settings_file_v6)
+        self.templar.render(template_data, metadata, self.settings_file_v6, None)
+
+
+    def restart_dhcp(self, service_name):
```

Comment:
Specify return type (if applicable) and the type of `service_name` if it is a built-in.